Fix preventing addon RCs that dont have version and similar prefixes from being deleted

vjsamuel opened the pull request:
Fixes https://github.com/kubernetes/kubernetes/issues/20125

vjsamuel committed:
Fix preventing addon RCs that dont have version and similar prefixes from being deleted

Removing unused variable

Removing the update-addons() and for_update variable

bgrant0607 commented:
cc @mikedanese @fgrzadkowski

sjenning committed:
remove unused function

fgrzadkowski commented:
@k8s-bot ok to test

fgrzadkowski commented:
@marekbiskup (author of this code)

AFAIU we were using this trick with suffixes to perform updates on RC. However, current implementation is just stopping and re-creating it. That means two things:
* You can delete all the update-related code we have
* Stop & delete for a particular addon can be spread in time, e.g. for a given addon we might delete it as the first one, but create it as the last one. That might cause some downtime (potentially long). Maybe we can at least try to sort objects by name to alleviate this? WDYT?

At the same time I think we should just start using Deployments for add-ons and just forget about this code. @nikhiljindal Is Deployment object ready to be used for cluster add-ons?

vjsamuel commented:
@fgrzadkowski i have gone ahead and removed the update related code and it should ideally be replaced with either based on version present inside of the spec or based on deployment objects.

fgrzadkowski commented:
@vjsamuel I don't see any new pushes. I still see update related code.

vjsamuel commented:
@fgrzadkowski there is only on commit and in that commit I have removed the update related code as marekbiskup  had suggested in his comment.

vjsamuel commented:
```
-                if [[ "${suffix_on_server}" == "${suffix_from_file}" ]]; then
 -                    nsnames_for_ignore="${nsnames_for_ignore} ${nsname_from_file}"
 -                else
 -                    for_update="${for_update} ${nsname_on_server};${addon_path}"
 -                fi
```

This is the piece of code which I have removed that does the version mismatch check to do the update of objects.

fgrzadkowski commented:
I walked through your commit again and I think it's a bit messy (that's why I missed your update-related change). From one side you're removing _some_ update related code (the one you've listed) but then leave the rest (various vars for updating or ignoring objects). What do you intend to do with object with the same name?

vjsamuel commented:
@fgrzadkowski Im sorry for the lack of clarity. The unused variable got missed out. The intent is to remove the concept of relying on version numbers to be used for updating of addons as per the description in #20125.

yujuhong committed:
kubelet: monitor the health of pleg

PLEG is reponsible for listing the pods running on the node. If it's hung
due to non-responsive container runtime or internal bugs, we should restart
kubelet.

vjsamuel commented:
I have removed the unused code in my last patch

yujuhong committed:
pleg: add an internal clock for testability

Also add tests for the health check.

david-mcmahon committed:
Pass latest or stable to build/push-official-release.sh.

wojtek-t committed:
Workaround long latency of POST pods

fgrzadkowski commented on cluster/saltbase/salt/kube-addons/kube-addon-update.sh:
```diff
@@ -330,22 +293,18 @@ function update-objects() {
 
     # output variables (globals)
     nsnames_for_delete=""
     for_update=""
```
How about `for_update`. I think we don't use it either. Function `update-objects` can also be removed if I understand your intentions correctly.

fgrzadkowski commented on cluster/saltbase/salt/kube-addons/kube-addon-update.sh:
```diff
@@ -434,17 +380,15 @@ function match-objects() {
+    match-objects ${addon_path} ${obj_type}
 
     log DBG "${obj_type}: nsnames_for_delete=${nsnames_for_delete}"
     log DBG "${obj_type}: for_update=${for_update}"
```
This can also be deleted.

fgrzadkowski commented on cluster/saltbase/salt/kube-addons/kube-addon-update.sh:
```diff
@@ -456,11 +400,6 @@ function reconcile-objects() {
     create-objects "${obj_type}" "${new_files}"
     update-objects "${obj_type}" "${for_update}"
```
This can also be deleted.

vjsamuel commented:
The intent conveyed is correct. I have removed the update objects as well.

fgrzadkowski commented on cluster/saltbase/salt/kube-addons/kube-addon-update.sh:
```diff
@@ -471,28 +406,6 @@ function reconcile-objects() {
     fi
 }
 
-function update-addons() {
```
I think you intended to delete 'update-objects' and 'update-object' not this method.

vjsamuel commented:
Im not sure how the additional commits came into my PR

vjsamuel commented:
Will open a fresh PR for this.
